jdaviz/core/loaders/resolvers/test_resolver.py

```diff
@@ -30,3 +35,163 @@ def test_server_is_remote_callback():
+
+    resolver.treat_table_as_query = True
+
+    assert resolver.parsed_input_is_query is True
+    assert resolver.observation_table_populated is True
+    assert 's_region' in resolver.observation_table.headers_avail
+
+    resolver.vue_link_by_wcs()
+    assert resolver.is_wcs_linked is True
+
+    resolver.toggle_custom_toolbar()
+    assert resolver.custom_toolbar_enabled is True
+    assert len(resolver._footprint_marks) == 3
+    assert len(resolver._footprint_groups) == 3
+    assert all(idx in resolver._footprint_groups for idx in range(3))
+
+    mark = resolver._footprint_marks[1]
+    px, py = np.mean(mark.x), np.mean(mark.y)
+    idx = find_closest_polygon_mark(px, py, resolver._footprint_marks)
+    assert idx is not None
+    assert 0 <= idx < 3
+
+    resolver.toggle_custom_toolbar()
+    assert resolver.custom_toolbar_enabled is False
+    assert resolver._footprint_marks == []
+    assert resolver._footprint_groups == {}
+
+
+def test_remove_footprints(imviz_helper):
+    wcs = WCS({
+        'CTYPE1': 'RA---TAN', 'CUNIT1': 'deg', 'CDELT1': -0.0002777777778,
+        'CRPIX1': 1, 'CRVAL1': 337.5202808,
+        'CTYPE2': 'DEC--TAN', 'CUNIT2': 'deg', 'CDELT2': 0.0002777777778,
+        'CRPIX2': 1, 'CRVAL2': -20.83333306,
+    })
+    arr = np.ones((200, 200))
+    ndd = NDData(arr, wcs=wcs)
+    imviz_helper.load_data(ndd, data_label='test_image')
+
+    table = Table()
+    table['Dataset'] = ['obs1']
+    table['s_region'] = [
+        'POLYGON 337.499 -20.831 337.501 -20.831 337.501 -20.829 337.499 -20.829'
+    ]
+
+    ldr = imviz_helper.loaders['object']
+    ldr.object = table
+    resolver = ldr._obj
+    resolver.treat_table_as_query = True
+    resolver.vue_link_by_wcs()
+
+    # Display footprints
+    resolver.toggle_custom_toolbar()
+    n_marks_before = len(resolver._footprint_marks)
+    assert n_marks_before == 1
+
+    # Remove footprints
+    resolver.toggle_custom_toolbar()
+
+    # ASsert marks are removed
+    assert len(resolver._footprint_marks) == 0
+    assert len(resolver._footprint_groups) == 0
+
+
+def test_multiselect(imviz_helper):
+    # Load data first
+    wcs = WCS({
+        'CTYPE1': 'RA---TAN', 'CUNIT1': 'deg', 'CDELT1': -0.0002777777778,
+        'CRPIX1': 1, 'CRVAL1': 337.5202808,
+        'CTYPE2': 'DEC--TAN', 'CUNIT2': 'deg', 'CDELT2': 0.0002777777778,
+        'CRPIX2': 1, 'CRVAL2': -20.83333306,
+    })
+    arr = np.ones((200, 200))
+    ndd = NDData(arr, wcs=wcs)
+    imviz_helper.load_data(ndd, data_label='test_image')
+
+    table = Table()
+    table['Dataset'] = ['obs1', 'obs2', 'obs3']
+    table['s_region'] = [
+        'POLYGON 337.499 -20.831 337.501 -20.831 337.501 -20.829 337.499 -20.829',
+        'POLYGON 337.502 -20.831 337.504 -20.831 337.504 -20.829 337.502 -20.829',
+        'POLYGON 337.505 -20.831 337.507 -20.831 337.507 -20.829 337.505 -20.829'
+    ]
+
+    ldr = imviz_helper.loaders['object']
+    ldr.object = table
+    resolver = ldr._obj
```

Comment:
(same as above - let's just use `ldr.` and `ldr._obj` so its obvious what is public vs internal